### PR TITLE
Ignore deleting pod from updating when resync

### PR DESCRIPTION
**Pull request by TommyLike**
For #384

Now volcano would add the deleting pod back into cache when resync, fix it by ignore it.

### CODE REVIEW

**Comment by TommyLike** [comment]
@hzxuzhonghu  please take a look

File: pkg/controllers/job/job_controller_resync.go [hunk 1]
```diff
@@ -69,20 +69,16 @@ func (cc *Controller) syncTask(oldTask *v1.Pod) error {
...
-			glog.V(3).Infof("Pod <%v/%v> was deleted, removed from cache.", oldTask.Namespace, oldTask.Name)
-
-			return nil
-		}
-		return fmt.Errorf("failed to get Pod <%v/%v>: err %v", oldTask.Namespace, oldTask.Name, err)
-	}
 
-	return cc.cache.UpdatePod(newPod)
+	//Delete from cache when pod does not exist or `DeletionTimestamp` not nil
+	if (err == nil && newPod.DeletionTimestamp != nil) || (err != nil && errors.IsNotFound(err)) {
```
Comment (hzxuzhonghu) [inline pkg/controllers/job/job_controller_resync.go]: I donot like this hacky way(check DeletionTimestamp here), it is already done in informer event handler.

Comment (TommyLike) [reply pkg/controllers/job/job_controller_resync.go]: Let's forget about the issue itself, would this be a more correct behavior for the `resync` logic?

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_resync.go]: not that i think, look at my analysis

**Comment by TommyLike** [comment]
Close for #384

**Pull request closed without merging**